pkg/reconciler/volumeclaim/pvchandler.go

```diff
@@ -77,15 +77,44 @@ func (c *defaultPVCHandler) CreatePVCFromVolumeClaimTemplate(ctx context.Context
 	case apierrors.IsNotFound(err):
 		_, err := c.clientset.CoreV1().PersistentVolumeClaims(claim.Namespace).Create(ctx, claim, metav1.CreateOptions{})
 		if err != nil {
-			if apierrors.IsAlreadyExists(err) {
+	_, getErr := c.clientset.CoreV1().PersistentVolumeClaims(claim.Namespace).Get(ctx, claim.Name, metav1.GetOptions{})
+	switch {
+	case apierrors.IsNotFound(getErr):
+		_, createErr := c.clientset.CoreV1().PersistentVolumeClaims(claim.Namespace).Create(ctx, claim, metav1.CreateOptions{})
+		if createErr == nil {
+			c.logger.Infof("Created PersistentVolumeClaim %s in namespace %s", claim.Name, claim.Namespace)
+			return nil
+		}
+
+		if apierrors.IsAlreadyExists(createErr) {
+			c.logger.Infof("Tried to create PersistentVolumeClaim %s in namespace %s, but it already exists",
+				claim.Name, claim.Namespace)
+			return nil
+		}
+
+		// This is a retry-able error
+		if apierrors.IsForbidden(createErr) && strings.Contains(createErr.Error(), "exceeded quota") {
+			return fmt.Errorf("%w: %v", ErrPvcCreationFailedRetryable, createErr.Error())
+		}
```

Comment:
Would it make sense to also include `apierrors.IsConflict()` if the conflict is not on the PVC itself? For example if there is a conflict updating a PVC Resource Quota

Reply:
I am not sure I fully follow 🤔